batcher/batcher_test.go

```diff
@@ -0,0 +1,105 @@
+package batcher
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb"
+)
+
+// TestBatch_Size ensures that a batcher generates a batch when the size threshold is reached.
+func TestBatch_Size(t *testing.T) {
+	batchSize := 5
+	batcher := NewBatcher(batchSize, time.Hour)
+	if batcher == nil {
+		t.Fatal("failed to create batcher for size test")
+	}
+
+	in := make(chan influxdb.Point)
+	out := make(chan []influxdb.Point)
+
+	go batcher.Start(in, out)
+
+	var p influxdb.Point
+	for i := 0; i < batchSize; i++ {
```

Comment:
I think the idea behind this test to show that it works in they way you want is to have line 27 `batch := <-out` block, which I'm sure it does.  But because you aren't launching the for loop in a go routine, it actually never waits.  It's immediately ready to run as the code had to artificially wait on the for loop.

So in this test, you aren't really making sure your channels really wait to flush because the for loop is blocking, not the channel read.